tmail-backend/guice/distributed/src/main/java/com/linagora/tmail/ScheduledReconnectionHandler.java

```diff
@@ -329,16 +349,26 @@ private Mono<Void> restart() {
                 .then());
     }
     
-    private boolean restartNeeded() {
+    public boolean restartNeeded() {
         return QUEUES_TO_MONITOR.stream()
             .filter(this::hasNoConsumers)
             .findAny()
             .isPresent();
     }
 
     private boolean hasNoConsumers(String queue) {
-        return mqManagementAPI.queueDetails(configuration.getVhost().orElse(DEFAULT_VHOST), queue)
-            .getConsumerDetails()
-            .isEmpty();
+        try {
+            boolean hasNoConsumers = mqManagementAPI.queueDetails(configuration.getVhost().orElse(DEFAULT_VHOST), queue)
+                .getConsumerDetails()
+                .isEmpty();
+
+            if (hasNoConsumers) {
+                LOGGER.warn("The {} queue has no consumers", queue);
+            }
+
+            return hasNoConsumers;
+        } catch (RabbitMQManagementAPI.QueueNotFoundException e) {
+            return false;
+        }
     }
```

Comment:
Double negation spotted.

Let's call the method `hasConsumer` instead and do the negation at the call site?